src/analyzer-range/analyzer-range.repository.ts

```diff
@@ -1,20 +1,22 @@
 import { Injectable } from '@nestjs/common';
-import { EntityManager, Repository } from 'typeorm';
+import { EntityManager, Repository, DataSource } from 'typeorm';
 
 import { AnalyzerRange } from '../entities/analyzer-range.entity';
-
+import { useSlaveQueryRunner } from '../utilities/use-slave-query';
 @Injectable()
 export class AnalyzerRangeRepository extends Repository<AnalyzerRange> {
-  constructor(entityManager: EntityManager) {
+  constructor(private readonly dataSource: DataSource, entityManager: EntityManager) {
     super(AnalyzerRange, entityManager);
   }
 
   async getAnalyzerRangesByCompIds(
     componentIds: string[],
   ): Promise<AnalyzerRange[]> {
-    return this.createQueryBuilder('ar')
-      .innerJoinAndSelect('ar.component', 'c')
-      .where('c.id IN (:...componentIds)', { componentIds })
-      .getMany();
+      return useSlaveQueryRunner(this.dataSource, async (qr) => {
```

Comment:
Instead of injecting the `dataSource`, this could just access the data source through the entity manager:
```suggestion
      return useSlaveQueryRunner(this.manager.connection, async (qr) => {
```

Reply:
I will update it with  this.manager.connection.